lib/mix/tasks/inertia.install.ex

```diff
@@ -0,0 +1,332 @@
+      ])
+
+    content = inertia_root_html()
+
+    Igniter.create_new_file(igniter, file_path, content, on_exists: :overwrite)
+  end
+
+  defp web_dir(igniter) do
+    igniter
+    |> Igniter.Libs.Phoenix.web_module()
+    |> inspect()
+    |> Macro.underscore()
+  end
+
+  defp inertia_root_html() do
+    """
+    <!DOCTYPE html>
+    <html lang="en">
+      <head>
+        <meta charset="utf-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <meta name="csrf-token" content={get_csrf_token()} />
+        <.inertia_title><%= assigns[:page_title] %></.inertia_title>
+        <.inertia_head content={@inertia_head} />
+        <link phx-track-static rel="stylesheet" href={~p"/assets/css/app.css"} />
+        <script type="module" defer phx-track-static src={~p"/assets/app.js"} />
+      </head>
+      <body>
+        {@inner_content}
+      </body>
+    </html>
+    """
+  end
+
+  @doc false
+  def update_esbuild_config(igniter) do
+    igniter
+    |> Igniter.Project.Config.configure(
+      "config.exs",
+      :esbuild,
+      [:version],
+      "0.21.5"
+    )
+    |> Igniter.Project.Config.configure(
+      "config.exs",
+      :esbuild,
+      [Igniter.Project.Application.app_name(igniter)],
+      {:code,
+       Sourceror.parse_string!("""
+       [
+        args:
+          ~w(js/app.jsx --bundle --chunk-names=chunks/[name]-[hash] --splitting --format=esm  --target=es2020 --outdir=../priv/static/assets --external:/fonts/* --external:/images/*),
+        cd: Path.expand("../assets", __DIR__),
+        env: %{"NODE_PATH" => Path.expand("../deps", __DIR__)}
+       ]
+       """)}
+    )
+    |> Igniter.add_task("esbuild.install")
+  end
+
+  @doc false
+  def setup_client(igniter) do
+    case igniter.args.options[:client_framework] do
+      "react" ->
+        igniter
+        |> install_client_package()
+        |> Igniter.create_new_file("assets/tsconfig.json", react_tsconfig_json(),
+          on_exists: :overwrite
+        )
+        |> Igniter.create_new_file("assets/js/app.jsx", inertia_app_jsx(), on_exists: :overwrite)
+
+      framework when framework in ["vue", "svelte"] ->
+        igniter
+        |> install_client_package()
+        |> Igniter.assign(:client_framework, framework)
+
+      _ ->
+        igniter
+    end
+  end
+
+  defp install_client_package(igniter) do
+    {packages, dev_packages} =
+      case igniter.args.options[:client_framework] do
+        "react" -> {"@inertiajs/react react react-dom", "@types/react"}
+        "vue" -> {"@inertiajs/vue3 vue vue-loader", "@vue/compiler-sfc vue-tsc typescript"}
+        "svelte" -> {"@inertiajs/svelte svelte", "svelte-loader svelte-preprocess typescript"}
+      end
+
+    igniter
+    |> Igniter.add_task("cmd", ["npm install --prefix assets #{packages}"])
+    |> Igniter.add_task("cmd", ["npm install --prefix assets --save-dev #{dev_packages}"])
+  end
+
+  defp react_tsconfig_json() do
+    """
+    {
```

Comment:
@derrickreimer `tsconfig.json` isn't strictly necessary to get up and running with inertia, would you want it to be included in the installer?

Reply:
In a perfect world, the user would be able to opt in to TypeScript from the installer (I'm personally a big fan of the Elixir + TypeScript combo). If that's a big lift though, we can start without it.